src/theory/strings/theory_strings_preprocess.cpp

```diff
@@ -105,52 +106,87 @@ Node StringsPreprocess::simplify( Node t, std::vector< Node > &new_nodes ) {
     Node lemma = NodeManager::currentNM()->mkNode( kind::ITE, cond, b1, b2 );
     new_nodes.push_back( lemma );
     retNode = skt;
-  } else if( t.getKind() == kind::STRING_STRIDOF ) {
-    Node sk2 = NodeManager::currentNM()->mkSkolem( "io2", NodeManager::currentNM()->stringType(), "created for indexof" );
-    Node sk3 = NodeManager::currentNM()->mkSkolem( "io3", NodeManager::currentNM()->stringType(), "created for indexof" );
-    Node sk4 = NodeManager::currentNM()->mkSkolem( "io4", NodeManager::currentNM()->stringType(), "created for indexof" );
+  }
+  else if (t.getKind() == kind::STRING_STRIDOF)
+  {
+    // processing term:  indexof( x, y, n )
+
     Node skk;
     if( options::stringUfReduct() ){
       skk = getUfAppForNode( kind::STRING_STRIDOF, t );
     }else{
-      skk = NodeManager::currentNM()->mkSkolem( "iok", NodeManager::currentNM()->integerType(), "created for indexof" );
+      skk = nm->mkSkolem("iok", nm->integerType(), "created for indexof");
     }
-    Node st = NodeManager::currentNM()->mkNode( kind::STRING_SUBSTR, t[0], t[2], NodeManager::currentNM()->mkNode( kind::MINUS, NodeManager::currentNM()->mkNode( kind::STRING_LENGTH, t[0] ), t[2] ) );
-    //TODO: simplify this (only applies when idof != -1)
-    Node eq = st.eqNode( NodeManager::currentNM()->mkNode( kind::STRING_CONCAT, sk2, sk3, sk4 ) );
-    new_nodes.push_back( eq );
-    
-    //learn range of idof?
-    Node negone = NodeManager::currentNM()->mkConst( ::CVC4::Rational(-1) );
-    Node krange = NodeManager::currentNM()->mkNode( kind::GEQ, skk, negone );
+
+    Node negone = nm->mkConst(::CVC4::Rational(-1));
+    Node krange = nm->mkNode(kind::GEQ, skk, negone);
+    // assert:   indeof( x, y, n ) >= -1
     new_nodes.push_back( krange );
-    krange = NodeManager::currentNM()->mkNode( kind::GT, NodeManager::currentNM()->mkNode( kind::STRING_LENGTH, t[0] ), skk);
+    krange = nm->mkNode(kind::GT, nm->mkNode(kind::STRING_LENGTH, t[0]), skk);
```

Comment:
Shouldn't this be GEQ? Because indexof("abc", "", 3) = 3.

Reply:
Yes, good catch.